### PR TITLE
Add header of the home screen

### PR DESCRIPTION
**Pull request by adrifdez**
Add header design to project

### CODE REVIEW

File: src/views/home/components/header/header.styles.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  },
+  "&:hover": {
+    backgroundColor: theme.palette.grey.light,
+  },
+});
+
+const useHeaderHomeStyles = createUseStyles((theme: Theme) => ({
+  header: {
+    ...flexBox,
+    justifyContent: "space-between",
```
Comment (amonsosanz) [inline src/views/home/components/header/header.styles.ts:30]: Why do we need the `justifyContent: "space-between"` and the `width: "100%"`?

File: src/views/home/components/header/header.styles.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    fontWeight: theme.fontWeights.medium,
+    "& > *": {
+      width: "100%",
```
Comment (amonsosanz) [inline src/views/home/components/header/header.styles.ts:33]: Why is this required? I mean, the logo and the subtitle will naturally get all the space they need.

File: src/views/home/components/header/header.styles.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  },
+  settingsBtn: {
+    ...flexBox,
+    alignItems: "center",
+    "& a": {
+      ...btn(theme),
+    },
+  },
+  myAccountBtn: {
+    ...flexBox,
```
Comment (amonsosanz) [inline src/views/home/components/header/header.styles.ts:44]: Given that these buttons are in a flex parent, instead of the `width: "100%"` I'd use `flex: 1`.

File: src/views/home/home.styles.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import { Theme } from "src/styles/theme";
+
+const useHomeStyles = createUseStyles((theme: Theme) => ({
+  chainRow: {
+    textAlign: "center",
+  },
+  chain: {
+    display: "inline-flex",
+    justifyContent: "center",
```
Comment (amonsosanz) [inline src/views/home/home.styles.ts:11]: Is this required?

File: src/views/shared/container/container.styles.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  root: {
+    width: "100%",
+    justifyContent: "center",
+    display: "flex",
+    backgroundColor: ({ background }: ContainerStyles) =>
+      background ? theme.palette.white : theme.palette.primary.main,
+  },
+  wrapper: {
+    width: "100%",
+    maxWidth: 700 + 2 * theme.spacing(3.5),
```
Comment (amonsosanz) [inline src/views/shared/container/container.styles.ts:19]: Why this computation?

File: src/views/home/components/header/header.styles.ts (hunk 1 above)
Comment (adrifdez) [reply src/views/home/components/header/header.styles.ts:30]: True I forgot remove it when I redesigned it

File: src/views/home/components/header/header.styles.ts (hunk 2 above)
Comment (adrifdez) [reply src/views/home/components/header/header.styles.ts:33]: it's need it to keep the logo in the real center with the Bridge text

File: src/views/home/home.styles.ts (hunk 4 above)
Comment (adrifdez) [reply src/views/home/home.styles.ts:11]: No because it's align it by the padding

File: src/views/shared/container/container.styles.ts (hunk 5 above)
Comment (adrifdez) [reply src/views/shared/container/container.styles.ts:19]: I brought this from `wallet-ui` because I didn't find any reference to the wrapper

File: src/views/home/components/header/header.styles.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  header: {
+    ...flexBox,
+    justifyContent: "space-between",
+    fontWeight: theme.fontWeights.medium,
+    "& > *": {
+      width: "100%",
+    },
+  },
+  settingsBtn: {
+    ...flexBox,
```
[the remark this answers is not in the document]
Comment (adrifdez) [reply src/views/home/components/header/header.styles.ts:36]: I prefer this way than create a new class and add two classes to `className` attribute but I'm open to any solution

[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/views/home/components/header/header.styles.ts:36]: Let's wait for @elias-garcia PoV

File: src/styles/theme.ts [hunk 7]
```diff
@@ -17,12 +17,12 @@ const theme = {
       hover: "#000411",
     },
     grey: {
-      veryLight: "#e1e1f1",
-      light: "#f3f3f8",
+      veryLight: "#f3f3f8",
```
Comment (elias-garcia) [inline src/styles/theme.ts:20]: I would remove all the colors that we are not using in the bridge by the moment. Apart from that I would try to put some order in this file.

File: src/styles/theme.ts [hunk 8]
```diff
@@ -17,12 +17,12 @@ const theme = {
...
+      light: "#E2E5EE",
```
Comment (elias-garcia) [inline src/styles/theme.ts:21]: Use lowercase for the hex colors to be consistent please.

File: src/routing/routes.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { Routes, Route } from "react-router-dom";
+
+import Home from "src/views/home/home.view";
+
+function Router(): JSX.Element {
```
Comment (elias-garcia) [inline src/routing/routes.tsx:5]: The file is named `routes.tsx` but the component is called `Router`, which is the correct one? `Router` looks better.

**Comment by elias-garcia** [comment]
The icon `ethIcon` should be renamed to `eth-icon.svg`, we don't use camel case for filenames.

File: src/views/home/components/header/header.styles.ts (hunk 1 above)
Comment (amonsosanz) [reply src/views/home/components/header/header.styles.ts]: But in this case, we don't need `flex: 1`, so instead of the spread I would add `display: "flex"`.

File: src/views/home/components/header/header.styles.ts (hunk 2 above)
Comment (amonsosanz) [reply src/views/home/components/header/header.styles.ts]: I may not get the point but I can't appreciate any difference:

bridge-logo-centering

File: src/views/shared/container/container.styles.ts (hunk 5 above)
Comment (amonsosanz) [reply src/views/shared/container/container.styles.ts:19]: > I didn't find any reference to the wrapper

I don't know what you mean by this.

The point is, if we're doing this computation, there must be a reason. I wonder why can't we just set `max-width: 756px;`.

File: src/views/home/components/header/header.styles.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (elias-garcia) [reply src/views/home/components/header/header.styles.ts:36]: I agree with Alex here. The way to abstract things is creating components, but not creating this style constants. If you want to abstract a button for example, just create a `Button` component. But I wouldn't do it by the moment, I would just inline the styles.

File: src/views/home/components/header/header.styles.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  "&:hover": {
+    backgroundColor: theme.palette.grey.light,
+  },
+});
+
+const useHeaderHomeStyles = createUseStyles((theme: Theme) => ({
+  header: {
+    ...flexBox,
+    fontWeight: theme.fontWeights.medium,
+    "& > *": {
```
Comment (elias-garcia) [inline src/views/home/components/header/header.styles.ts:31]: This kind of wildcards are very dangerous. Just apply the styles to the children that need it.

File: src/views/home/components/header/header.styles.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+      flex: 1,
+    },
+  },
+  settingsBtn: {
```
Comment (elias-garcia) [inline src/views/home/components/header/header.styles.ts:35]: Avoid this kind of abbreviations, `settingsButton` is more descriptive and readable.

File: src/views/home/components/transaction-actions/transaction-actions.styles.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+import { Theme } from "src/styles/theme";
+
+const useTransactionActionsStyles = createUseStyles((theme: Theme) => ({
+  buttons: {
+    display: "flex",
+    justifyContent: "center",
+    margin: [[theme.spacing(1), theme.spacing(5)]],
+  },
+  actionBtn: {
```
Comment (elias-garcia) [inline src/views/home/components/transaction-actions/transaction-actions.styles.ts:11]: Same as before, use `actionButton` please

File: src/views/home/components/transaction-actions/transaction-actions.view.tsx [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+import { ReactComponent as Deposit } from "src/assets/icons/deposit.svg";
+import { ReactComponent as Withdraw } from "src/assets/icons/withdraw.svg";
+import useTransactionActionsStyles from "src/views/home/components/transaction-actions/transaction-actions.styles";
+
+const TransactionActions = () => {
+  const classes = useTransactionActionsStyles();
+  return (
+    <div className={classes.buttons}>
+      <Link to={"/"} className={classes.actionBtn}>
+        <p className={classes.circle}>
```
Comment (elias-garcia) [inline src/views/home/components/transaction-actions/transaction-actions.view.tsx:12]: Why the `circle` is a `<p>` element?

File: src/views/home/components/header/header.styles.ts [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    ...flexBox,
+    alignItems: "center",
+    "& a": {
+      ...btn(theme),
+    },
+  },
+  myAccountBtn: {
```
Comment (elias-garcia) [inline src/views/home/components/header/header.styles.ts:42]: `myAccountButton`

File: src/views/home/components/header/header.styles.ts [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+    },
+  },
+  logo: {
+    alignItems: "center",
+    textAlign: "center",
+    flexDirection: "column",
+    "& svg": {
+      width: 43,
+    },
+    "& p": {
```
Comment (elias-garcia) [inline src/views/home/components/header/header.styles.ts:57]: Avoid targeting inner elements, what if we change from a `p` to an `h1`? It's always better to have an specific class for each element, like `logoText`. Far more robust to changes.

**Review by elias-garcia — CHANGES_REQUESTED** [review]

File: src/views/shared/container/container.styles.ts (hunk 5 above)
Comment (adrifdez) [reply src/views/shared/container/container.styles.ts:19]: Ahh to be equal than the padding

Comment (elias-garcia) [reply src/views/shared/container/container.styles.ts:19]: This was a tricky thing from the wallet... I can't remember why but it has a reason. But we can remove it from here and set it as Alex said.

File: src/views/home/components/header/header.styles.ts (hunk 2 above)
Comment (adrifdez) [reply src/views/home/components/header/header.styles.ts]: Peek 2022-03-04 13-44

File: src/views/home/components/transaction-actions/transaction-actions.view.tsx (hunk 13 above)
Comment (adrifdez) [reply src/views/home/components/transaction-actions/transaction-actions.view.tsx:12]: because Link is an `a` element

File: src/views/home/components/header/header.styles.ts (hunk 2 above)
Comment (amonsosanz) [reply src/views/home/components/header/header.styles.ts]: Aha! it's a browser-specific issue!

**Review by amonsosanz — APPROVED** [review]

File: src/views/home/components/header/header.styles.ts [hunk 16]
```diff
@@ -0,0 +1,56 @@
+import { createUseStyles } from "react-jss";
+
+import { Theme } from "src/styles/theme";
+
+const useHeaderHomeStyles = createUseStyles((theme: Theme) => ({
```
Comment (elias-garcia) [inline src/views/home/components/header/header.styles.ts:5]: This variable should be called `useHeaderStyles`

File: src/views/home/components/transaction-actions/transaction-actions.styles.ts (hunk 12 above)
Comment (elias-garcia) [reply src/views/home/components/transaction-actions/transaction-actions.styles.ts:11]: I still see `actionButton` here

File: src/views/home/components/header/header.view.tsx [hunk 17]
```diff
@@ -0,0 +1,32 @@
+import { Link } from "react-router-dom";
+
+import { ReactComponent as Logo } from "src/assets/hermez-logo.svg";
+import { ReactComponent as Settings } from "src/assets/icons/settings.svg";
```
Comment (elias-garcia) [inline src/views/home/components/header/header.view.tsx:4]: When you import an icon it would be great to add the `Icon` prefix:

```
import { ReactComponent as SettingsIcon } from "src/assets/icons/settings.svg";
import { ReactComponent as QRIcon } from "src/assets/icons/qr.svg";
```

File: src/styles/theme.ts [hunk 18]
```diff
@@ -2,57 +2,25 @@ const theme = {
...
-      main: "#8248e5",
-      dark: "#6d00f1",
+      light: "#e2e5ee",
+      hover: "rgba(67, 85, 140, 0.1)",
     },
-    green: "#219653",
   },
   hoverTransition: "all 100ms",
   fontWeights: {
-    normal: "400",
```
Comment (elias-garcia) [inline src/styles/theme.ts:45]: We can keep the `fontWeights` map as we are going to need it. At least I would keep this:

```
fontWeights: {
    normal: "400",
    medium: "500",
    bold: "700",
}
```

File: src/views/home/components/transaction-actions/transaction-actions.view.tsx [hunk 19]
```diff
@@ -0,0 +1,27 @@
+import { Link } from "react-router-dom";
+
+import { ReactComponent as Deposit } from "src/assets/icons/deposit.svg";
```
Comment (elias-garcia) [inline src/views/home/components/transaction-actions/transaction-actions.view.tsx:3]:
```ts
import { ReactComponent as DepositIcon } from "src/assets/icons/deposit.svg";
import { ReactComponent as WithdrawIcon } from "src/assets/icons/withdraw.svg";
```

**Review by elias-garcia — CHANGES_REQUESTED** [review]

File: src/views/home/components/transaction-actions/transaction-actions.view.tsx (hunk 13 above)
Comment (elias-garcia) [reply src/views/home/components/transaction-actions/transaction-actions.view.tsx:12]: I know, but a <p> element is for paragraphs. Why you don't apply the circle styles to the `a` element and you keep just the icon inside?

File: src/styles/theme.ts (hunk 18 above)
Comment (adrifdez) [reply src/styles/theme.ts:45]: I'll add just bold because normal is by default

**Review by elias-garcia — APPROVED** [review]
